### ISSUE TITLE
Haxe Completion

### ISSUE THREAD

**Issue by ncannasse** [fdorg/flashdevelop#1064]
FYI I noticed today that adding -D display-details was not correctly handled in Haxe compiler, which was triggering a full recompilation when using the compiler cache.
I have fixed it on Haxe GIT https://github.com/HaxeFoundation/haxe/commit/22633b6f0649fb870347f372c7931fc6962ce9dc but it might be better if you could always add -D display-details when compiling _and_ asking completion, so there is no mismatch of configuration between the two. It seems the order of defines did also matters, which I have fixed as well.

**Comment by elsassph** [comment]
You're talking about the compiler server right?

Gee that may be tricky as there are many situations to handle: 

- build fully handled by FD: using project properties the build command is constructed,
- build/completion based on a user hxml file
- completion based on a user hxml but build as a custom command
- completion based on a 3rd party toolchain (openfl...) but build done buy the toolchain.

Only the first case can be handled correctly, others will be difficult to solve.

**Comment by ncannasse** [comment]
Yes, it's completion server related.

I think the only thing you need to do is to have ``--each -D display-details`` for each compilation/completion, instead of simply adding `-D display-details` for completion.

The other completion-specific parameter (-D use_rtti_doc) is alrady ignored in Haxe 3.x

**Comment by elsassph** [comment]
`--each -D display-details  --next ...`?

**Comment by ncannasse** [comment]
yes. Also, it seems you're using --each debug as well, so you need to group them together or else the second each will reset flags

**Comment by elsassph** [comment]
And if you have your own `--each` in your additional compiler options?

**Comment by ncannasse** [comment]
Then we need for fix that in haxe compiler. I guess we could change so they get added together instead of reset. Pinging @Simn on the issue

**Comment by Simn** [comment]
I can change it but I'm not sure if there might be problems due to duplicate command line arguments in some cases.
